doc/bum_storm_control/bum_storm_control_hld.md

```diff
@@ -0,0 +1,398 @@
+ 3. Support debug commands as described in "Debug Commands" section below.
+
+
+### 1.1.3 Scalability Requirements
+Storm-control configuration is done on physical ports. Hence it should be supported on all physical ports of the system. 
+
+### 1.1.4 Warm Boot Requirements
+Storm-control functionality should continue to work across warm reboot. 
+
+- To support planned system warm boot.
+- To support SWSS docker warm boot.
+
+
+## 1.2 Design Overview
+
+
+### 1.2.1 Basic Approach
+The Policer Orchestration agent is used for handling BUM storm-control configuration. 
+
+### 1.2.2 Container
+No new containers are introduced. Functionality is added to the existing swss container. 
+
+### 1.2.3 SAI Overview
+The existing SAI policer and interface APIs shall be used to support this feature and they are available at below location, 
+https://github.com/opencomputeproject/SAI/blob/master/inc/saipolicer.h <br>
+https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h
+
+
+The details of the changes will be discussed in the design section below.
+
+
+# 2 Functionality
+Refer to section 1
+## 2.1 Target Deployment Use Cases
+Any Layer-2 network segment where unknown-unicast / broadcast / unknown-multicast traffic storm should be  controlled. 
+
+
+## 2.2 Functional Description
+Refer to section 1.1
+
+## 2.3 Limitations
+BUM storm control 
+- Configuration is not supported on VLAN and port-channel interfaces. User can configure on physical port which is part of a VLAN / port-channel.
+- Statistics is not supported. 
+- REST, gNMI and Klish CLI are not supported. 
+ 
+
+
+# 3 Design
+## 3.1 Overview
+
+
+![Storm Control](images/storm_control_architecture.png "Figure 1: Storm control High Level Architecture")
+
+
+__Figure 1: Storm Control High Level Architecture__
+
+1) Storm-control configurations are parsed and stored in CFG_PORT_STORM_CONTROL_TABLE in Configuration database by the Management Framework.
+2) The Policer Orchestration Agent subscribes to notifications from the CFG_PORT_STORM_CONTROL_TABLE and parses the input parameters (interface, storm-control type, kbps). A policer_name is created internally by encoding the interface_name and storm_control_type.
+3) create_policer SAI API is invoked to create a policer with the given input parameters. 
+4) The identifier of the policer created is associated with the encoded policer_name.
+5) The policer identifier is passed to set_port_attribute SAI API to set the appropriate type of storm-control on the port. 
+
+
+## 3.2 DB Changes
+This section describes the changes made to different DBs for supporting storm-control.
+
+
+### 3.2.1 CONFIG_DB
+A new table CFG_PORT_STORM_CONTROL_TABLE is introduced in the configuration database for the purpose of storing storm-control configuration parameters. This table is filled by the management framework. 
+#### CFG_PORT_STORM_CONTROL_TABLE
```

Comment:
missing yang model

Reply:
https://github.com/Azure/sonic-buildimage/pull/7355

Reply:
Please correct the table name as PORT_STORM_CONTROL and use "|" has key separators in config-DB.

Reply:
Updated Table Name.